src/main/java/teammates/ui/webapi/CreateFeedbackSessionAction.java

```diff
@@ -74,4 +84,47 @@ JsonResult execute() {
         return new JsonResult(output);
     }
 
+    private void createFeedbackQuestions(String copyCourseId, String newCourseId, String feedbackSessionName,
+            String oldSessionName) {
+        logic.getFeedbackQuestionsForSession(oldSessionName, copyCourseId).forEach(question -> {
+            FeedbackQuestionAttributes attributes = FeedbackQuestionAttributes.builder()
+                    .withCourseId(newCourseId)
+                    .withFeedbackSessionName(feedbackSessionName)
+                    .withGiverType(question.getGiverType())
+                    .withRecipientType(question.getRecipientType())
+                    .withQuestionNumber(question.getQuestionNumber())
+                    .withNumberOfEntitiesToGiveFeedbackTo(question.getNumberOfEntitiesToGiveFeedbackTo())
+                    .withShowResponsesTo(question.getShowResponsesTo())
+                    .withShowGiverNameTo(question.getShowGiverNameTo())
+                    .withShowRecipientNameTo(question.getShowRecipientNameTo())
+                    .withQuestionDetails(question.getQuestionDetails())
+                    .withQuestionDescription(question.getQuestionDescription())
+                    .build();
+
+            // validate questions (giver & recipient)
```

Comment:
let's think about this for a second - why do we need to validate again if the old questions have already been validated? Tell me a scenario in which we run into a validation error here

Reply:
I removed the part that checks for the validity of giver and recipient.
If a feedback session is copied to a course with different number of students / teams, the `numOfGeneratedMsqChoices` would be different. So the part that validates question details is needed to update `numOfGeneratedMsqChoices` .

Reply:
@damithc I need to raise this here. This is a problem I've been aware of for quite some time. (hello #8453)

This part here validates that the max or min number of choices (whichever ones configured) is/are within the range of the number of generated choices (student, team, instructor, etc.). At the first glance, it seems sensible. However, this creates an unnecessary coupling between the question and the course structure.

Consider this:
- A question with min. 5 choice, generated from instructors, is created, in a course with 6 instructors
- 2 instructors are removed from that course
- This question is now unanswerable, as the validation is done only during question creation/update, but not when the coupled course structure is changed

And in this very example (copying course), any MSQ question with minimum number of selected choice generated from students/teams is not copiable, because a new course doesn't have any student list to be generated from!

I suggest to just remove this validation.

Reply:
> I suggest to just remove this validation. Not because this validation creates the problem or removing this validation solves the problem, but because this validation doesn't solve the problem it intends to.

I agree. Let's do that.